### PR TITLE
More control over wave sending

### PR DESCRIPTION
**Pull request by MEEPofFaith**
- New rule: `waveSending`. If set to false, the send wave button is disabled.
- `Spawn Wave` command changed: Can now be set to spawn a natural wave (The same as the send wave button)

https://user-images.githubusercontent.com/54301439/186617519-1b83c8e8-cd7e-4e14-8dea-213e5d13d6a9.mp4

Can be useful for world procs sending some waves (with drop zone kill), then later giving the control to the player.

If your pull request is **not** translation or serverlist-related, read the list of requirements below and check each box:

- [X] I have read the [contribution guidelines](https://github.com/Anuken/Mindustry/blob/master/CONTRIBUTING.md).
- [X] I have ensured that my code compiles, if applicable.
- [X] I have ensured that any new features in this PR function correctly in-game, if applicable.

### CODE REVIEW

File: core/src/mindustry/logic/WaveType.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package mindustry.logic;
+
+public enum WaveType{
```
Comment (Anuken) [inline core/src/mindustry/logic/WaveType.java:3]: An enum with two values is pointless, just use a boolean instead.

**Comment by MEEPofFaith** [comment]
https://user-images.githubusercontent.com/54301439/186734375-bafea54a-49dc-4f2b-87b6-d5283136d0b3.mp4

~~You know, I feel like there should be a selector for booleans similar to enum selection~~

**Comment by MEEPofFaith** [comment]
https://user-images.githubusercontent.com/54301439/186806347-2e5397af-61f2-4a1a-bbbb-cc4aef25b98c.mp4

**Comment by Anuken** [comment]
I don't see the point of making these instructions accept hard-coded booleans instead of variables, and you've almost certainly broken a lot of existing world logic with this change without good reason.

**Comment by MEEPofFaith** [comment]
I mean there are some parameters in logic commands that only accept booleans, but fair enough.
That could be left to a separate PR later if people get tired of manually typing "true" or "false."

**Comment by Anuken** [comment]
> That could be left to a separate PR later if people get tired of manually typing "true" or "false."

That would be a very unreasonable thing to get tired of, I am forced to do it all the time in Java and it's fine.

File: core/src/mindustry/logic/LStatements.java [hunk 2]
```diff
@@ -1291,16 +1291,21 @@ public LCategory category(){
 
     @RegisterStatement("spawnwave")
     public static class SpawnWaveStatement extends LStatement{
+        public String natural = "false";
```
Comment (Anuken) [inline core/src/mindustry/logic/LStatements.java:1294]: Would be more backwards-compatible to have this as a last parameter so existing logic doesn't break.

**Pull request merged**